### ISSUE TITLE
WebImageDrop and Admin rights doesn't work

### ISSUE THREAD

**Issue by greg6614** [mgarin/weblaf#322]
Hi,

First I want congratulate you for WebLAF. It's amazing and really beautiful, but I've a problem. For more convenience I would use my application directly from an executable .EXE. So I used some tools which allow me to do this but when I bundle my .JAR in an executable .EXE with admin rights, I'm unable to drop image in WebImageDrop.

If this problem is known and if you know how to solve it, that's will be great.

**Comment by Endogen** [comment]
I guess it's important to know which Java wrapper you use to test it

**Comment by greg6614** [comment]
Indeed, I tried to use Launch4J, Advanced Installer 12.1, JavaExe and Jar2Exe 2.0 and still the same problem.

**Comment by mgarin** [comment]
I've seen this issue a few times on different Windows versions, sometimes even when run from IDE/jar, sometimes in installed application. Unfortunately I was not able to narrow down the source of this issue as it appeared randomly for me and (as far as I remember) rebooting Windows solved this problem whenever it began to reproduce, so I felt that it might be some native JDK/Windows-related bug.

A few questions:
- What JDK/Windows versions do you have?
- Do you have the same problem when running from IDE?
- Can you try this simple Swing drop code in IDE/exe? Does it work for you? code:
```java
public class DropExample
{
    public static void main ( final String[] args )
    {
        final JFrame frame = new JFrame ();

        frame.setLayout ( new BorderLayout () );

        final JLabel dropArea = new JLabel ( "DROP HERE" );
        dropArea.setTransferHandler ( new TransferHandler ()
        {
            @Override
            public boolean canImport ( final JComponent comp, final DataFlavor[] transferFlavors )
            {
                return true;
            }

            @Override
            public boolean importData ( final JComponent comp, final Transferable t )
            {
                try
                {
                    final String data = ( String ) t.getTransferData ( DataFlavor.stringFlavor );
                    dropArea.setText ( data );
                    return true;
                }
                catch ( final Throwable e )
                {
                    return false;
                }
            }
        } );
        frame.add ( dropArea );

        frame.setSize ( 600, 600 );
        frame.setDefaultCloseOperation ( JFrame.EXIT_ON_CLOSE );
        frame.setLocationRelativeTo ( null );
        frame.setVisible ( true );
    }
}
```

**Comment by greg6614** [comment]
Thanks for your reply, I'm using JDK 8 Update 45 with Windows 7. This issue doesn't appear when running directly from IDE or Jar but only when I wrap my Jar with admin rights. Your code works perfectly with IDE but doesn't work in a wrapped jar with Launch4J with admin rights but works with normal user rights.

**Comment by mgarin** [comment]
> Thanks for your reply, I'm using JDK 8 Update 45 with Windows 7. This issue doesn't appear when running directly from IDE or Jar but only when I wrap my Jar with admin rights.

Thanks for this information, I will try reproducing this issue.

> Your code works perfectly with IDE but doesn't work in a wrapped jar with Launch4J with admin rights but works with normal user rights.

The one I have posted?
Because if it is so - it means that any D&D simply doesn't work under specified circumstances.
Meaning that this is basically a general Swing or even JDK issue.

---

The only time I actually managed to reproduce this issue in debug in IDE (was actually trying to get it) gave me nothing, because drop operation simply did not reach Swing code. It looked like native window was simply denying D&D operation within the window. 

And since it has something to do with permissions application has - probably you just can't drop anything from other applications that doesn't have admin rights?
